Scroller/EditableWrapper.js

```diff
@@ -623,45 +657,38 @@ const EditableWrapper = (props) => {
 			completeEditingByKeyDown();
 			mutableRef.current.needToPreventEvent = true;
 		} else if (is('left', keyCode) || is('right', keyCode)) {
+			const nextTarget = getTargetByDirectionFromElement(getDirection(keyCode), target);
+
 			if (selectedItem) {
+				// If keyDown event target is item(=not button), move item.
 				if (target.getAttribute('role') !== 'button') {
-					if (Number(selectedItem.style.order) - 1 < mutableRef.current.hideIndex) {
-						if (repeat) {
-							SpotlightAccelerator.processKey(ev, moveItemsByKeyDown);
-						} else {
-							SpotlightAccelerator.reset();
-							moveItemsByKeyDown(ev);
-						}
-					}
-					ev.preventDefault();
-					ev.stopPropagation();
-				}
-			} else {
-				const nextTarget = getTargetByDirectionFromElement(getDirection(keyCode), target);
+					handleMoveItemsByKeyDown(ev, repeat);
+				// If keyDown event target is button and next spot target is item, move item and then spot selected item.
+				} else if (nextTarget?.getAttribute('role') !== 'button') {
+					handleMoveItemsByKeyDown(ev, repeat);
 
-				// Check if focus leaves scroll container.
-				if (nextTarget && !getContainersForNode(nextTarget).includes(mutableRef.current.spotlightId) && !repeat) {
-					reset();
+					setPointerMode(false);
+					Spotlight.focus(selectedItem.children[1]);
+				// If keyDown event target is button and next spot target is button, check whether focus leaves the scroll container.
+				} else {
+					// Set lastInputType to 'key' to prevent `handleMoveItemsByScroll` from being executed unexpectedly.
+					mutableRef.current.lastInputType = 'key';
```

Comment:
I understand why we need this as it looks not easy to cover the case regarding global handlers and normal handlers. But, still I'm wonder whether we're able to gather the cases of setting `lastInputType` to 'key'. 😉 In general, the state cycle of `lastInputType` is not as same as our other code since it does not purely depend on the last input type but which key is pressed for key inputs.